crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -156,6 +164,23 @@ impl IdentityManager {
             "Handling a keys query response"
         );
 
+        // Parse the strings into server names and filter out our own server. We should
+        // never get failures from our own server but let's remove it as a
+        // precaution anyways.
+        let failed_servers = response
+            .failures
+            .keys()
+            .filter_map(|k| ServerName::parse(k).ok())
+            .filter(|s| s != self.user_id().server_name());
+        let successful_servers = response.device_keys.keys().map(|u| u.server_name());
+
+        // Append the new failed servers and remove any successful servers. We remove
+        // the successful servers explicitly so we keep incrementing the delay
+        // on the failed servers instead of removing them when the entry times
+        // out.
```

Comment:
```suggestion
        // Append the new failed servers and remove any successful servers. We
        // need to explicitly remove the successful servers because the cache
        // doesn't automatically remove entries that elapse. Instead, the effect
        // is that elapsed servers will be retried and their delays incremented.
```